edx_rest_api_client/client.py

```diff
@@ -103,10 +105,20 @@ def __init__(self, url, signing_key=None, username=None, full_name=None, email=N
         session = session or requests.Session()
         session.headers['User-Agent'] = self.user_agent()
 
+        if retry_attempts:
+            # Retries should only be attempted in the cases where it is okay for the caller to be able to blocking
+            # it's thread. For example, if a request is in an atomic transaction, we do not want to attempt a retry.
```

Comment:
I don't think it is just an atomic transaction.  It would be from any synchronous call to a real user.  Also, for celery tasks, we could also get in trouble if there are many of them. :)